pytket/qir/conversion/api.py

```diff
@@ -124,6 +107,49 @@ def pytket_to_qir(
             assert not "unsupported return type"  # type: ignore
 
 
+def _check_circuit_before_conversion(
+    circuit: Circuit,
+    int_type: int = 64,
+) -> None:
+    if len(circuit.q_registers) > 1 or (
+        len(circuit.q_registers) == 1 and circuit.q_registers[0].name != "q"
+    ):
+        raise ValueError(
+            """The circuit that should be converted should only have the default
+            quantum register. You can convert it using the pytket
+            compiler pass `FlattenRelabelRegistersPass`."""
+        )
+
+    if int_type != 32 and int_type != 64:
+        raise ValueError("the integer size must be 32 or 64")
+
+    for creg in circuit.c_registers:
+        if creg.size > int_type:
+            raise ValueError(
+                f"""classical registers must not have more than {int_type} bits, \
+you could try to set cut_pytket_register=True in the conversion"""
+            )
+
+
+def check_circuit(
+    circuit: Circuit,
+    int_type: int = 64,
+) -> None:
+    """runs the checks for the conversion. Running this check
+     is recomended for big circuits that take a long time to be converted.
+    This functions will throw error with suggestion on how to resolve the
+    problems.
+    :param circuit: given circuit
+    :param int_type: size of each integer, allowed value 32 and 64
+    """
+    _check_circuit_before_conversion(circuit, int_type)
+
+    set_circ_register = set([creg.name for creg in circuit.c_registers])
+    for b in set([b.reg_name for b in circuit.bits]):
+        if b not in set_circ_register:
+            raise ValueError(f"Used register {b} in not a valid register")
```

Comment:
I think this is because of the circuit structure I am testing, it has a lot of classic registers. Happy to do another measurement of the runtime if you think that would be good.

Reply:
I'd be interested to see the numbers. If it's almost always small compared with the time taken for `pytket_to_qir()` then I'd be inclined to put all the checks into that function.

Reply:
The difference was only showing up for the big circuit, I have not checked the runtime for the small circuits, but I would expect that the difference is below 0.001 for them.
I will run the profiling for both cases again to make sure there was nothing interfering with the measurements last time.
What do you mean with `put all the checks into that function`?

Reply:
> What do you mean with `put all the checks into that function`?

I mean, have `pytket_qir()` do all sanity checks early, by calling the `check_circuit()` function. (We wouldn't then need the underscored function which it calls currently.)

Reply:
I have run a detailed profile for the check function, the runtime for the additional register check is ~1s now. I am not sure what changed to the other run I did.

I have clean up the check function, it is now always fully run in the conversion.

See [66f7f02](https://github.com/CQCL/pytket-qir/pull/147/commits/66f7f02ab4f176ac4e74a35eda71ba7598d87bb4)